src/explore-education-statistics-admin/src/pages/edit-release/ReleaseDataPage.tsx

```diff
@@ -1,38 +1,273 @@
+                  {dataFile.title}
+                </SummaryListItem>
+                <SummaryListItem term="Data file">
+                  <a
+                    href={service.createDownloadDataFileLink(
+                      releaseId,
+                      dataFile.file.id,
+                    )}
+                  >
+                    {dataFile.file.fileName}
+                  </a>
+                </SummaryListItem>
+                <SummaryListItem term="Filesize">
+                  {dataFile.fileSize.size.toLocaleString()}{' '}
+                  {dataFile.fileSize.unit}
+                </SummaryListItem>
+                <SummaryListItem term="Number of rows">
+                  {dataFile.numberOfRows.toLocaleString()}
+                </SummaryListItem>
+                <SummaryListItem term="Metadata file">
+                  <a
+                    href={service.createDownloadDataMetadataFileLink(
+                      releaseId,
+                      dataFile.file.id,
+                    )}
+                  >
+                    {dataFile.metadataFile.fileName}
+                  </a>
+                </SummaryListItem>
+                <SummaryListItem
+                  term="Actions"
+                  actions={
+                    <Link
+                      to="#"
+                      onClick={_ => setDeleteFilesRequest(dataFile.file.id)}
+                    >
+                      Delete files
+                    </Link>
+                  }
+                />
+              </SummaryList>
+            ))}
+          <Formik<FormValues>
+            enableReinitialize
+            initialValues={{
+              subjectTitle: '',
+              dataFile: null,
+              metadataFile: null,
+            }}
+            onSubmit={async (values: FormValues, actions) => {
+              service
+                .uploadDataFiles(releaseId, {
+                  subjectTitle: values.subjectTitle,
+                  dataFile: values.dataFile as File,
+                  metadataFile: values.metadataFile as File,
+                })
+                .then(_ => service.getReleaseDataFiles(releaseId))
+                .then(setDataFiles)
+                .then(_ => {
+                  actions.resetForm();
+                  document
+                    .querySelectorAll(`#${formId} input[type='file']`)
+                    .forEach(input => {
+                      const fileInput = input as HTMLInputElement;
+                      fileInput.value = '';
+                    });
+                });
+            }}
+            validationSchema={Yup.object<FormValues>({
+              subjectTitle: Yup.string().required('Enter a subject title'),
+              dataFile: Yup.mixed().required('Choose a data file'),
+              metadataFile: Yup.mixed().required('Choose a metadata file'),
+            })}
+            render={(form: FormikProps<FormValues>) => {
```

Comment:
I don't know enough about react - but why do we have a render method AND the html above?

Reply:
This is a special Formik method that renders just the Form that it controls.